// react-csspin/dist/index.js
import * as jr from "react";
function Ar(a) {
  return a && a.__esModule && Object.prototype.hasOwnProperty.call(a, "default") ? a.default : a;
}
function Ir(a) {
  if (a.__esModule)
    return a;
  var S = a.default;
  if (typeof S == "function") {
    var E = function s() {
      return this instanceof s ? Reflect.construct(S, arguments, this.constructor) : S.apply(this, arguments);
    };
    E.prototype = S.prototype;
  } else
    E = {};
  return Object.defineProperty(E, "__esModule", { value: !0 }), Object.keys(a).forEach(function(s) {
    var O = Object.getOwnPropertyDescriptor(a, s);
    Object.defineProperty(E, s, O.get ? O : {
      enumerable: !0,
      get: function() {
        return a[s];
      }
    });
  }), E;
}
var ke = { exports: {} }, be = {};
const ur = /* @__PURE__ */ Ir(jr);
/**
 * @license React
 * react-jsx-runtime.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Ke;
function $r() {
  return Ke || (Ke = 1, process.env.NODE_ENV !== "production" && function() {
    var a = ur, S = Symbol.for("react.element"), E = Symbol.for("react.portal"), s = Symbol.for("react.fragment"), O = Symbol.for("react.strict_mode"), l = Symbol.for("react.profiler"), j = Symbol.for("react.provider"), d = Symbol.for("react.context"), b = Symbol.for("react.forward_ref"), w = Symbol.for("react.suspense"), f = Symbol.for("react.suspense_list"), g = Symbol.for("react.memo"), k = Symbol.for("react.lazy"), F = Symbol.for("react.offscreen"), z = Symbol.iterator, M = "@@iterator";
    function V(e) {
      if (e === null || typeof e != "object")
        return null;
      var r = z && e[z] || e[M];
      return typeof r == "function" ? r : null;
    }
    var U = a.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
    function Y(e) {
      {
        for (var r = arguments.length, n = new Array(r > 1 ? r - 1 : 0), c = 1; c < r; c++)
          n[c - 1] = arguments[c];
        N("error", e, n);
      }
    }
    function N(e, r, n) {
      {
        var c = U.ReactDebugCurrentFrame, A = c.getStackAddendum();
        A !== "" && (r += "%s", n = n.concat([A]));
        var D = n.map(function(C) {
          return String(C);
        });
        D.unshift("Warning: " + r), Function.prototype.apply.call(console[e], console, D);
      }
    }
    var L = !1, i = !1, ee = !1, de = !1, ve = !1, oe;
    oe = Symbol.for("react.module.reference");
    function pe(e) {
      return !!(typeof e == "string" || typeof e == "function" || e === s || e === l || ve || e === O || e === w || e === f || de || e === F || L || i || ee || typeof e == "object" && e !== null && (e.$$typeof === k || e.$$typeof === g || e.$$typeof === j || e.$$typeof === d || e.$$typeof === b || // This needs to include all possible module reference object
      // types supported by any Flight configuration anywhere since
      // we don't know which Flight build this will end up being used
      // with.
      e.$$typeof === oe || e.getModuleId !== void 0));
    }
    function ie(e, r, n) {
      var c = e.displayName;
      if (c)
        return c;
      var A = r.displayName || r.name || "";
      return A !== "" ? n + "(" + A + ")" : n;
    }
    function ue(e) {
      return e.displayName || "Context";
    }
    function H(e) {
      if (e == null)
        return null;
      if (typeof e.tag == "number" && Y("Received an unexpected object in getComponentNameFromType(). This is likely a bug in React. Please file an issue."), typeof e == "function")
        return e.displayName || e.name || null;
      if (typeof e == "string")
        return e;
      switch (e) {
        case s:
          return "Fragment";
        case E:
          return "Portal";
        case l:
          return "Profiler";
        case O:
          return "StrictMode";
        case w:
          return "Suspense";
        case f:
          return "SuspenseList";
      }
      if (typeof e == "object")
        switch (e.$$typeof) {
          case d:
            var r = e;
            return ue(r) + ".Consumer";
          case j:
            var n = e;
            return ue(n._context) + ".Provider";
          case b:
            return ie(e, e.render, "ForwardRef");
          case g:
            var c = e.displayName || null;
            return c !== null ? c : H(e.type) || "Memo";
          case k: {
            var A = e, D = A._payload, C = A._init;
            try {
              return H(C(D));
            } catch {
              return null;
            }
          }
        }
      return null;
    }
    var G = Object.assign, re = 0, K, Q, ne, se, t, u, m;
    function h() {
    }
    h.__reactDisabledLog = !0;
    function v() {
      {
        if (re === 0) {
          K = console.log, Q = console.info, ne = console.warn, se = console.error, t = console.group, u = console.groupCollapsed, m = console.groupEnd;
          var e = {
            configurable: !0,
            enumerable: !0,
            value: h,
            writable: !0
          };
          Object.defineProperties(console, {
            info: e,
            log: e,
            warn: e,
            error: e,
            group: e,
            groupCollapsed: e,
            groupEnd: e
          });
        }
        re++;
      }
    }
    function P() {
      {
        if (re--, re === 0) {
          var e = {
            configurable: !0,
            enumerable: !0,
            writable: !0
          };
          Object.defineProperties(console, {
            log: G({}, e, {
              value: K
            }),
            info: G({}, e, {
              value: Q
            }),
            warn: G({}, e, {
              value: ne
            }),
            error: G({}, e, {
              value: se
            }),
            group: G({}, e, {
              value: t
            }),
            groupCollapsed: G({}, e, {
              value: u
            }),
            groupEnd: G({}, e, {
              value: m
            })
          });
        }
        re < 0 && Y("disabledDepth fell below zero. This is a bug in React. Please file an issue.");
      }
    }
    var p = U.ReactCurrentDispatcher, y;
    function T(e, r, n) {
      {
        if (y === void 0)
          try {
            throw Error();
          } catch (A) {
            var c = A.stack.trim().match(/\n( *(at )?)/);
            y = c && c[1] || "";
          }
        return `
` + y + e;
      }
    }
    var x = !1, _;
    {
      var B = typeof WeakMap == "function" ? WeakMap : Map;
      _ = new B();
    }
    function o(e, r) {
      if (!e || x)
        return "";
      {
        var n = _.get(e);
        if (n !== void 0)
          return n;
      }
      var c;
      x = !0;
      var A = Error.prepareStackTrace;
      Error.prepareStackTrace = void 0;
      var D;
      D = p.current, p.current = null, v();
      try {
        if (r) {
          var C = function() {
            throw Error();
          };
          if (Object.defineProperty(C.prototype, "props", {
            set: function() {
              throw Error();
            }
          }), typeof Reflect == "object" && Reflect.construct) {
            try {
              Reflect.construct(C, []);
            } catch (te) {
              c = te;
            }
            Reflect.construct(e, [], C);
          } else {
            try {
              C.call();
            } catch (te) {
              c = te;
            }
            e.call(C.prototype);
          }
        } else {
          try {
            throw Error();
          } catch (te) {
            c = te;
          }
          e();
        }
      } catch (te) {
        if (te && c && typeof te.stack == "string") {
          for (var R = te.stack.split(`
`), J = c.stack.split(`
`), W = R.length - 1, q = J.length - 1; W >= 1 && q >= 0 && R[W] !== J[q]; )
            q--;
          for (; W >= 1 && q >= 0; W--, q--)
            if (R[W] !== J[q]) {
              if (W !== 1 || q !== 1)
                do
                  if (W--, q--, q < 0 || R[W] !== J[q]) {
                    var Z = `
` + R[W].replace(" at new ", " at ");
                    return e.displayName && Z.includes("<anonymous>") && (Z = Z.replace("<anonymous>", e.displayName)), typeof e == "function" && _.set(e, Z), Z;
                  }
                while (W >= 1 && q >= 0);
              break;
            }
        }
      } finally {
        x = !1, p.current = D, P(), Error.prepareStackTrace = A;
      }
      var le = e ? e.displayName || e.name : "", He = le ? T(le) : "";
      return typeof e == "function" && _.set(e, He), He;
    }
    function X(e, r, n) {
      return o(e, !1);
    }
    function ce(e) {
      var r = e.prototype;
      return !!(r && r.isReactComponent);
    }
    function ae(e, r, n) {
      if (e == null)
        return "";
      if (typeof e == "function")
        return o(e, ce(e));
      if (typeof e == "string")
        return T(e);
      switch (e) {
        case w:
          return T("Suspense");
        case f:
          return T("SuspenseList");
      }
      if (typeof e == "object")
        switch (e.$$typeof) {
          case b:
            return X(e.render);
          case g:
            return ae(e.type, r, n);
          case k: {
            var c = e, A = c._payload, D = c._init;
            try {
              return ae(D(A), r, n);
            } catch {
            }
          }
        }
      return "";
    }
    var me = Object.prototype.hasOwnProperty, Fe = {}, Ye = U.ReactDebugCurrentFrame;
    function Ee(e) {
      if (e) {
        var r = e._owner, n = ae(e.type, e._source, r ? r.type : null);
        Ye.setExtraStackFrame(n);
      } else
        Ye.setExtraStackFrame(null);
    }
    function fr(e, r, n, c, A) {
      {
        var D = Function.call.bind(me);
        for (var C in e)
          if (D(e, C)) {
            var R = void 0;
            try {
              if (typeof e[C] != "function") {
                var J = Error((c || "React class") + ": " + n + " type `" + C + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + typeof e[C] + "`.This often happens because of typos such as `PropTypes.function` instead of `PropTypes.func`.");
                throw J.name = "Invariant Violation", J;
              }
              R = e[C](r, C, c, n, null, "SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED");
            } catch (W) {
              R = W;
            }
            R && !(R instanceof Error) && (Ee(A), Y("%s: type specification of %s `%s` is invalid; the type checker function must return `null` or an `Error` but returned a %s. You may have forgotten to pass an argument to the type checker creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and shape all require an argument).", c || "React class", n, C, typeof R), Ee(null)), R instanceof Error && !(R.message in Fe) && (Fe[R.message] = !0, Ee(A), Y("Failed %s type: %s", n, R.message), Ee(null));
          }
      }
    }
    var lr = Array.isArray;
    function _e(e) {
      return lr(e);
    }
    function dr(e) {
      {
        var r = typeof Symbol == "function" && Symbol.toStringTag, n = r && e[Symbol.toStringTag] || e.constructor.name || "Object";
        return n;
      }
    }
    function vr(e) {
      try {
        return We(e), !1;
      } catch {
        return !0;
      }
    }
    function We(e) {
      return "" + e;
    }
    function Le(e) {
      if (vr(e))
        return Y("The provided key is an unsupported type %s. This value must be coerced to a string before before using it here.", dr(e)), We(e);
    }
    var ye = U.ReactCurrentOwner, pr = {
      key: !0,
      ref: !0,
      __self: !0,
      __source: !0
    }, qe, Ue, Re;
    Re = {};
    function yr(e) {
      if (me.call(e, "ref")) {
        var r = Object.getOwnPropertyDescriptor(e, "ref").get;
        if (r && r.isReactWarning)
          return !1;
      }
      return e.ref !== void 0;
    }
    function br(e) {
      if (me.call(e, "key")) {
        var r = Object.getOwnPropertyDescriptor(e, "key").get;
        if (r && r.isReactWarning)
          return !1;
      }
      return e.key !== void 0;
    }
    function hr(e, r) {
      if (typeof e.ref == "string" && ye.current && r && ye.current.stateNode !== r) {
        var n = H(ye.current.type);
        Re[n] || (Y('Component "%s" contains the string ref "%s". Support for string refs will be removed in a future major release. This case cannot be automatically converted to an arrow function. We ask you to manually fix this case by using useRef() or createRef() instead. Learn more about using refs safely here: https://reactjs.org/link/strict-mode-string-ref', H(ye.current.type), e.ref), Re[n] = !0);
      }
    }
    function mr(e, r) {
      {
        var n = function() {
          qe || (qe = !0, Y("%s: `key` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", r));
        };
        n.isReactWarning = !0, Object.defineProperty(e, "key", {
          get: n,
          configurable: !0
        });
      }
    }
    function Er(e, r) {
      {
        var n = function() {
          Ue || (Ue = !0, Y("%s: `ref` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", r));
        };
        n.isReactWarning = !0, Object.defineProperty(e, "ref", {
          get: n,
          configurable: !0
        });
      }
    }
    var gr = function(e, r, n, c, A, D, C) {
      var R = {
        // This tag allows us to uniquely identify this as a React Element
        $$typeof: S,
        // Built-in properties that belong on the element
        type: e,
        key: r,
        ref: n,
        props: C,
        // Record the component responsible for creating this element.
        _owner: D
      };
      return R._store = {}, Object.defineProperty(R._store, "validated", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: !1
      }), Object.defineProperty(R, "_self", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: c
      }), Object.defineProperty(R, "_source", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: A
      }), Object.freeze && (Object.freeze(R.props), Object.freeze(R)), R;
    };
    function Tr(e, r, n, c, A) {
      {
        var D, C = {}, R = null, J = null;
        n !== void 0 && (Le(n), R = "" + n), br(r) && (Le(r.key), R = "" + r.key), yr(r) && (J = r.ref, hr(r, A));
        for (D in r)
          me.call(r, D) && !pr.hasOwnProperty(D) && (C[D] = r[D]);
        if (e && e.defaultProps) {
          var W = e.defaultProps;
          for (D in W)
            C[D] === void 0 && (C[D] = W[D]);
        }
        if (R || J) {
          var q = typeof e == "function" ? e.displayName || e.name || "Unknown" : e;
          R && mr(C, q), J && Er(C, q);
        }
        return gr(e, R, J, A, c, ye.current, C);
      }
    }
    var Oe = U.ReactCurrentOwner, Ne = U.ReactDebugCurrentFrame;
    function fe(e) {
      if (e) {
        var r = e._owner, n = ae(e.type, e._source, r ? r.type : null);
        Ne.setExtraStackFrame(n);
      } else
        Ne.setExtraStackFrame(null);
    }
    var Se;
    Se = !1;
    function Pe(e) {
      return typeof e == "object" && e !== null && e.$$typeof === S;
    }
    function Ve() {
      {
        if (Oe.current) {
          var e = H(Oe.current.type);
          if (e)
            return `

Check the render method of \`` + e + "`.";
        }
        return "";
      }
    }
    function _r(e) {
      {
        if (e !== void 0) {
          var r = e.fileName.replace(/^.*[\\\/]/, ""), n = e.lineNumber;
          return `

Check your code at ` + r + ":" + n + ".";
        }
        return "";
      }
    }
    var ze = {};
    function Rr(e) {
      {
        var r = Ve();
        if (!r) {
          var n = typeof e == "string" ? e : e.displayName || e.name;
          n && (r = `

Check the top-level render call using <` + n + ">.");
        }
        return r;
      }
    }
    function Be(e, r) {
      {
        if (!e._store || e._store.validated || e.key != null)
          return;
        e._store.validated = !0;
        var n = Rr(r);
        if (ze[n])
          return;
        ze[n] = !0;
        var c = "";
        e && e._owner && e._owner !== Oe.current && (c = " It was passed a child from " + H(e._owner.type) + "."), fe(e), Y('Each child in a list should have a unique "key" prop.%s%s See https://reactjs.org/link/warning-keys for more information.', n, c), fe(null);
      }
    }
    function Je(e, r) {
      {
        if (typeof e != "object")
          return;
        if (_e(e))
          for (var n = 0; n < e.length; n++) {
            var c = e[n];
            Pe(c) && Be(c, r);
          }
        else if (Pe(e))
          e._store && (e._store.validated = !0);
        else if (e) {
          var A = V(e);
          if (typeof A == "function" && A !== e.entries)
            for (var D = A.call(e), C; !(C = D.next()).done; )
              Pe(C.value) && Be(C.value, r);
        }
      }
    }
    function Or(e) {
      {
        var r = e.type;
        if (r == null || typeof r == "string")
          return;
        var n;
        if (typeof r == "function")
          n = r.propTypes;
        else if (typeof r == "object" && (r.$$typeof === b || // Note: Memo only checks outer props here.
        // Inner props are checked in the reconciler.
        r.$$typeof === g))
          n = r.propTypes;
        else
          return;
        if (n) {
          var c = H(r);
          fr(n, e.props, "prop", c, e);
        } else if (r.PropTypes !== void 0 && !Se) {
          Se = !0;
          var A = H(r);
          Y("Component %s declared `PropTypes` instead of `propTypes`. Did you misspell the property assignment?", A || "Unknown");
        }
        typeof r.getDefaultProps == "function" && !r.getDefaultProps.isReactClassApproved && Y("getDefaultProps is only used on classic React.createClass definitions. Use a static property named `defaultProps` instead.");
      }
    }
    function Sr(e) {
      {
        for (var r = Object.keys(e.props), n = 0; n < r.length; n++) {
          var c = r[n];
          if (c !== "children" && c !== "key") {
            fe(e), Y("Invalid prop `%s` supplied to `React.Fragment`. React.Fragment can only have `key` and `children` props.", c), fe(null);
            break;
          }
        }
        e.ref !== null && (fe(e), Y("Invalid attribute `ref` supplied to `React.Fragment`."), fe(null));
      }
    }
    function Ge(e, r, n, c, A, D) {
      {
        var C = pe(e);
        if (!C) {
          var R = "";
          (e === void 0 || typeof e == "object" && e !== null && Object.keys(e).length === 0) && (R += " You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.");
          var J = _r(A);
          J ? R += J : R += Ve();
          var W;
          e === null ? W = "null" : _e(e) ? W = "array" : e !== void 0 && e.$$typeof === S ? (W = "<" + (H(e.type) || "Unknown") + " />", R = " Did you accidentally export a JSX literal instead of a component?") : W = typeof e, Y("React.jsx: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: %s.%s", W, R);
        }
        var q = Tr(e, r, n, A, D);
        if (q == null)
          return q;
        if (C) {
          var Z = r.children;
          if (Z !== void 0)
            if (c)
              if (_e(Z)) {
                for (var le = 0; le < Z.length; le++)
                  Je(Z[le], e);
                Object.freeze && Object.freeze(Z);
              } else
                Y("React.jsx: Static children should always be an array. You are likely explicitly calling React.jsxs or React.jsxDEV. Use the Babel transform instead.");
            else
              Je(Z, e);
        }
        return e === s ? Sr(q) : Or(q), q;
      }
    }
    function Pr(e, r, n) {
      return Ge(e, r, n, !0);
    }
    function Cr(e, r, n) {
      return Ge(e, r, n, !1);
    }
    var wr = Cr, xr = Pr;
    be.Fragment = s, be.jsx = wr, be.jsxs = xr;
  }()), be;
}
var he = {};
/**
 * @license React
 * react-jsx-runtime.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Xe;
function kr() {
  if (Xe)
    return he;
  Xe = 1;
  var a = ur, S = Symbol.for("react.element"), E = Symbol.for("react.fragment"), s = Object.prototype.hasOwnProperty, O = a.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.ReactCurrentOwner, l = { key: !0, ref: !0, __self: !0, __source: !0 };
  function j(d, b, w) {
    var f, g = {}, k = null, F = null;
    w !== void 0 && (k = "" + w), b.key !== void 0 && (k = "" + b.key), b.ref !== void 0 && (F = b.ref);
    for (f in b)
      s.call(b, f) && !l.hasOwnProperty(f) && (g[f] = b[f]);
    if (d && d.defaultProps)
      for (f in b = d.defaultProps, b)
        g[f] === void 0 && (g[f] = b[f]);
    return { $$typeof: S, type: d, key: k, ref: F, props: g, _owner: O.current };
  }
  return he.Fragment = E, he.jsx = j, he.jsxs = j, he;
}
process.env.NODE_ENV === "production" ? ke.exports = kr() : ke.exports = $r();
var ge = ke.exports, De = { exports: {} }, Ce, Ze;
function Me() {
  if (Ze)
    return Ce;
  Ze = 1;
  var a = "SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED";
  return Ce = a, Ce;
}
var we, Qe;
function Dr() {
  if (Qe)
    return we;
  Qe = 1;
  var a = Me();
  function S() {
  }
  function E() {
  }
  return E.resetWarningCache = S, we = function() {
    function s(j, d, b, w, f, g) {
      if (g !== a) {
        var k = new Error(
          "Calling PropTypes validators directly is not supported by the `prop-types` package. Use PropTypes.checkPropTypes() to call them. Read more at http://fb.me/use-check-prop-types"
        );
        throw k.name = "Invariant Violation", k;
      }
    }
    s.isRequired = s;
    function O() {
      return s;
    }
    var l = {
      array: s,
      bigint: s,
      bool: s,
      func: s,
      number: s,
      object: s,
      string: s,
      symbol: s,
      any: s,
      arrayOf: O,
      element: s,
      elementType: s,
      instanceOf: O,
      node: s,
      objectOf: O,
      oneOf: O,
      oneOfType: O,
      shape: O,
      exact: O,
      checkPropTypes: E,
      resetWarningCache: S
    };
    return l.PropTypes = l, l;
  }, we;
}
var Te = { exports: {} }, I = {};
/** @license React v16.13.1
 * react-is.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var er;
function Mr() {
  return er || (er = 1, process.env.NODE_ENV !== "production" && function() {
    var a = typeof Symbol == "function" && Symbol.for, S = a ? Symbol.for("react.element") : 60103, E = a ? Symbol.for("react.portal") : 60106, s = a ? Symbol.for("react.fragment") : 60107, O = a ? Symbol.for("react.strict_mode") : 60108, l = a ? Symbol.for("react.profiler") : 60114, j = a ? Symbol.for("react.provider") : 60109, d = a ? Symbol.for("react.context") : 60110, b = a ? Symbol.for("react.async_mode") : 60111, w = a ? Symbol.for("react.concurrent_mode") : 60111, f = a ? Symbol.for("react.forward_ref") : 60112, g = a ? Symbol.for("react.suspense") : 60113, k = a ? Symbol.for("react.suspense_list") : 60120, F = a ? Symbol.for("react.memo") : 60115, z = a ? Symbol.for("react.lazy") : 60116, M = a ? Symbol.for("react.block") : 60121, V = a ? Symbol.for("react.fundamental") : 60117, U = a ? Symbol.for("react.responder") : 60118, Y = a ? Symbol.for("react.scope") : 60119;
    function N(o) {
      return typeof o == "string" || typeof o == "function" || // Note: its typeof might be other than 'symbol' or 'number' if it's a polyfill.
      o === s || o === w || o === l || o === O || o === g || o === k || typeof o == "object" && o !== null && (o.$$typeof === z || o.$$typeof === F || o.$$typeof === j || o.$$typeof === d || o.$$typeof === f || o.$$typeof === V || o.$$typeof === U || o.$$typeof === Y || o.$$typeof === M);
    }
    function L(o) {
      if (typeof o == "object" && o !== null) {
        var X = o.$$typeof;
        switch (X) {
          case S:
            var ce = o.type;
            switch (ce) {
              case b:
              case w:
              case s:
              case l:
              case O:
              case g:
                return ce;
              default:
                var ae = ce && ce.$$typeof;
                switch (ae) {
                  case d:
                  case f:
                  case z:
                  case F:
                  case j:
                    return ae;
                  default:
                    return X;
                }
            }
          case E:
            return X;
        }
      }
    }
    var i = b, ee = w, de = d, ve = j, oe = S, pe = f, ie = s, ue = z, H = F, G = E, re = l, K = O, Q = g, ne = !1;
    function se(o) {
      return ne || (ne = !0, console.warn("The ReactIs.isAsyncMode() alias has been deprecated, and will be removed in React 17+. Update your code to use ReactIs.isConcurrentMode() instead. It has the exact same API.")), t(o) || L(o) === b;
    }
    function t(o) {
      return L(o) === w;
    }
    function u(o) {
      return L(o) === d;
    }
    function m(o) {
      return L(o) === j;
    }
    function h(o) {
      return typeof o == "object" && o !== null && o.$$typeof === S;
    }
    function v(o) {
      return L(o) === f;
    }
    function P(o) {
      return L(o) === s;
    }
    function p(o) {
      return L(o) === z;
    }
    function y(o) {
      return L(o) === F;
    }
    function T(o) {
      return L(o) === E;
    }
    function x(o) {
      return L(o) === l;
    }
    function _(o) {
      return L(o) === O;
    }
    function B(o) {
      return L(o) === g;
    }
    I.AsyncMode = i, I.ConcurrentMode = ee, I.ContextConsumer = de, I.ContextProvider = ve, I.Element = oe, I.ForwardRef = pe, I.Fragment = ie, I.Lazy = ue, I.Memo = H, I.Portal = G, I.Profiler = re, I.StrictMode = K, I.Suspense = Q, I.isAsyncMode = se, I.isConcurrentMode = t, I.isContextConsumer = u, I.isContextProvider = m, I.isElement = h, I.isForwardRef = v, I.isFragment = P, I.isLazy = p, I.isMemo = y, I.isPortal = T, I.isProfiler = x, I.isStrictMode = _, I.isSuspense = B, I.isValidElementType = N, I.typeOf = L;
  }()), I;
}
var $ = {};
/** @license React v16.13.1
 * react-is.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var rr;
function Fr() {
  if (rr)
    return $;
  rr = 1;
  var a = typeof Symbol == "function" && Symbol.for, S = a ? Symbol.for("react.element") : 60103, E = a ? Symbol.for("react.portal") : 60106, s = a ? Symbol.for("react.fragment") : 60107, O = a ? Symbol.for("react.strict_mode") : 60108, l = a ? Symbol.for("react.profiler") : 60114, j = a ? Symbol.for("react.provider") : 60109, d = a ? Symbol.for("react.context") : 60110, b = a ? Symbol.for("react.async_mode") : 60111, w = a ? Symbol.for("react.concurrent_mode") : 60111, f = a ? Symbol.for("react.forward_ref") : 60112, g = a ? Symbol.for("react.suspense") : 60113, k = a ? Symbol.for("react.suspense_list") : 60120, F = a ? Symbol.for("react.memo") : 60115, z = a ? Symbol.for("react.lazy") : 60116, M = a ? Symbol.for("react.block") : 60121, V = a ? Symbol.for("react.fundamental") : 60117, U = a ? Symbol.for("react.responder") : 60118, Y = a ? Symbol.for("react.scope") : 60119;
  function N(i) {
    if (typeof i == "object" && i !== null) {
      var ee = i.$$typeof;
      switch (ee) {
        case S:
          switch (i = i.type, i) {
            case b:
            case w:
            case s:
            case l:
            case O:
            case g:
              return i;
            default:
              switch (i = i && i.$$typeof, i) {
                case d:
                case f:
                case z:
                case F:
                case j:
                  return i;
                default:
                  return ee;
              }
          }
        case E:
          return ee;
      }
    }
  }
  function L(i) {
    return N(i) === w;
  }
  return $.AsyncMode = b, $.ConcurrentMode = w, $.ContextConsumer = d, $.ContextProvider = j, $.Element = S, $.ForwardRef = f, $.Fragment = s, $.Lazy = z, $.Memo = F, $.Portal = E, $.Profiler = l, $.StrictMode = O, $.Suspense = g, $.isAsyncMode = function(i) {
    return L(i) || N(i) === b;
  }, $.isConcurrentMode = L, $.isContextConsumer = function(i) {
    return N(i) === d;
  }, $.isContextProvider = function(i) {
    return N(i) === j;
  }, $.isElement = function(i) {
    return typeof i == "object" && i !== null && i.$$typeof === S;
  }, $.isForwardRef = function(i) {
    return N(i) === f;
  }, $.isFragment = function(i) {
    return N(i) === s;
  }, $.isLazy = function(i) {
    return N(i) === z;
  }, $.isMemo = function(i) {
    return N(i) === F;
  }, $.isPortal = function(i) {
    return N(i) === E;
  }, $.isProfiler = function(i) {
    return N(i) === l;
  }, $.isStrictMode = function(i) {
    return N(i) === O;
  }, $.isSuspense = function(i) {
    return N(i) === g;
  }, $.isValidElementType = function(i) {
    return typeof i == "string" || typeof i == "function" || i === s || i === w || i === l || i === O || i === g || i === k || typeof i == "object" && i !== null && (i.$$typeof === z || i.$$typeof === F || i.$$typeof === j || i.$$typeof === d || i.$$typeof === f || i.$$typeof === V || i.$$typeof === U || i.$$typeof === Y || i.$$typeof === M);
  }, $.typeOf = N, $;
}
var tr;
function sr() {
  return tr || (tr = 1, process.env.NODE_ENV === "production" ? Te.exports = Fr() : Te.exports = Mr()), Te.exports;
}
/*
object-assign
(c) Sindre Sorhus
@license MIT
*/
var xe, nr;
function Yr() {
  if (nr)
    return xe;
  nr = 1;
  var a = Object.getOwnPropertySymbols, S = Object.prototype.hasOwnProperty, E = Object.prototype.propertyIsEnumerable;
  function s(l) {
    if (l == null)
      throw new TypeError("Object.assign cannot be called with null or undefined");
    return Object(l);
  }
  function O() {
    try {
      if (!Object.assign)
        return !1;
      var l = new String("abc");
      if (l[5] = "de", Object.getOwnPropertyNames(l)[0] === "5")
        return !1;
      for (var j = {}, d = 0; d < 10; d++)
        j["_" + String.fromCharCode(d)] = d;
      var b = Object.getOwnPropertyNames(j).map(function(f) {
        return j[f];
      });
      if (b.join("") !== "0123456789")
        return !1;
      var w = {};
      return "abcdefghijklmnopqrst".split("").forEach(function(f) {
        w[f] = f;
      }), Object.keys(Object.assign({}, w)).join("") === "abcdefghijklmnopqrst";
    } catch {
      return !1;
    }
  }
  return xe = O() ? Object.assign : function(l, j) {
    for (var d, b = s(l), w, f = 1; f < arguments.length; f++) {
      d = Object(arguments[f]);
      for (var g in d)
        S.call(d, g) && (b[g] = d[g]);
      if (a) {
        w = a(d);
        for (var k = 0; k < w.length; k++)
          E.call(d, w[k]) && (b[w[k]] = d[w[k]]);
      }
    }
    return b;
  }, xe;
}
var je, ar;
function cr() {
  return ar || (ar = 1, je = Function.call.bind(Object.prototype.hasOwnProperty)), je;
}
var Ae, or;
function Wr() {
  if (or)
    return Ae;
  or = 1;
  var a = function() {
  };
  if (process.env.NODE_ENV !== "production") {
    var S = Me(), E = {}, s = cr();
    a = function(l) {
      var j = "Warning: " + l;
      typeof console < "u" && console.error(j);
      try {
        throw new Error(j);
      } catch {
      }
    };
  }
  function O(l, j, d, b, w) {
    if (process.env.NODE_ENV !== "production") {
      for (var f in l)
        if (s(l, f)) {
          var g;
          try {
            if (typeof l[f] != "function") {
              var k = Error(
                (b || "React class") + ": " + d + " type `" + f + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + typeof l[f] + "`.This often happens because of typos such as `PropTypes.function` instead of `PropTypes.func`."
              );
              throw k.name = "Invariant Violation", k;
            }
            g = l[f](j, f, b, d, null, S);
          } catch (z) {
            g = z;
          }
          if (g && !(g instanceof Error) && a(
            (b || "React class") + ": type specification of " + d + " `" + f + "` is invalid; the type checker function must return `null` or an `Error` but returned a " + typeof g + ". You may have forgotten to pass an argument to the type checker creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and shape all require an argument)."
          ), g instanceof Error && !(g.message in E)) {
            E[g.message] = !0;
            var F = w ? w() : "";
            a(
              "Failed " + d + " type: " + g.message + (F ?? "")
            );
          }
        }
    }
  }
  return O.resetWarningCache = function() {
    process.env.NODE_ENV !== "production" && (E = {});
  }, Ae = O, Ae;
}
var Ie, ir;
function Lr() {
  if (ir)
    return Ie;
  ir = 1;
  var a = sr(), S = Yr(), E = Me(), s = cr(), O = Wr(), l = function() {
  };
  process.env.NODE_ENV !== "production" && (l = function(d) {
    var b = "Warning: " + d;
    typeof console < "u" && console.error(b);
    try {
      throw new Error(b);
    } catch {
    }
  });
  function j() {
    return null;
  }
  return Ie = function(d, b) {
    var w = typeof Symbol == "function" && Symbol.iterator, f = "@@iterator";
    function g(t) {
      var u = t && (w && t[w] || t[f]);
      if (typeof u == "function")
        return u;
    }
    var k = "<<anonymous>>", F = {
      array: U("array"),
      bigint: U("bigint"),
      bool: U("boolean"),
      func: U("function"),
      number: U("number"),
      object: U("object"),
      string: U("string"),
      symbol: U("symbol"),
      any: Y(),
      arrayOf: N,
      element: L(),
      elementType: i(),
      instanceOf: ee,
      node: pe(),
      objectOf: ve,
      oneOf: de,
      oneOfType: oe,
      shape: ue,
      exact: H
    };
    function z(t, u) {
      return t === u ? t !== 0 || 1 / t === 1 / u : t !== t && u !== u;
    }
    function M(t, u) {
      this.message = t, this.data = u && typeof u == "object" ? u : {}, this.stack = "";
    }
    M.prototype = Error.prototype;
    function V(t) {
      if (process.env.NODE_ENV !== "production")
        var u = {}, m = 0;
      function h(P, p, y, T, x, _, B) {
        if (T = T || k, _ = _ || y, B !== E) {
          if (b) {
            var o = new Error(
              "Calling PropTypes validators directly is not supported by the `prop-types` package. Use `PropTypes.checkPropTypes()` to call them. Read more at http://fb.me/use-check-prop-types"
            );
            throw o.name = "Invariant Violation", o;
          } else if (process.env.NODE_ENV !== "production" && typeof console < "u") {
            var X = T + ":" + y;
            !u[X] && // Avoid spamming the console because they are often not actionable except for lib authors
            m < 3 && (l(
              "You are manually calling a React.PropTypes validation function for the `" + _ + "` prop on `" + T + "`. This is deprecated and will throw in the standalone `prop-types` package. You may be seeing this warning due to a third-party PropTypes library. See https://fb.me/react-warning-dont-call-proptypes for details."
            ), u[X] = !0, m++);
          }
        }
        return p[y] == null ? P ? p[y] === null ? new M("The " + x + " `" + _ + "` is marked as required " + ("in `" + T + "`, but its value is `null`.")) : new M("The " + x + " `" + _ + "` is marked as required in " + ("`" + T + "`, but its value is `undefined`.")) : null : t(p, y, T, x, _);
      }
      var v = h.bind(null, !1);
      return v.isRequired = h.bind(null, !0), v;
    }
    function U(t) {
      function u(m, h, v, P, p, y) {
        var T = m[h], x = K(T);
        if (x !== t) {
          var _ = Q(T);
          return new M(
            "Invalid " + P + " `" + p + "` of type " + ("`" + _ + "` supplied to `" + v + "`, expected ") + ("`" + t + "`."),
            { expectedType: t }
          );
        }
        return null;
      }
      return V(u);
    }
    function Y() {
      return V(j);
    }
    function N(t) {
      function u(m, h, v, P, p) {
        if (typeof t != "function")
          return new M("Property `" + p + "` of component `" + v + "` has invalid PropType notation inside arrayOf.");
        var y = m[h];
        if (!Array.isArray(y)) {
          var T = K(y);
          return new M("Invalid " + P + " `" + p + "` of type " + ("`" + T + "` supplied to `" + v + "`, expected an array."));
        }
        for (var x = 0; x < y.length; x++) {
          var _ = t(y, x, v, P, p + "[" + x + "]", E);
          if (_ instanceof Error)
            return _;
        }
        return null;
      }
      return V(u);
    }
    function L() {
      function t(u, m, h, v, P) {
        var p = u[m];
        if (!d(p)) {
          var y = K(p);
          return new M("Invalid " + v + " `" + P + "` of type " + ("`" + y + "` supplied to `" + h + "`, expected a single ReactElement."));
        }
        return null;
      }
      return V(t);
    }
    function i() {
      function t(u, m, h, v, P) {
        var p = u[m];
        if (!a.isValidElementType(p)) {
          var y = K(p);
          return new M("Invalid " + v + " `" + P + "` of type " + ("`" + y + "` supplied to `" + h + "`, expected a single ReactElement type."));
        }
        return null;
      }
      return V(t);
    }
    function ee(t) {
      function u(m, h, v, P, p) {
        if (!(m[h] instanceof t)) {
          var y = t.name || k, T = se(m[h]);
          return new M("Invalid " + P + " `" + p + "` of type " + ("`" + T + "` supplied to `" + v + "`, expected ") + ("instance of `" + y + "`."));
        }
        return null;
      }
      return V(u);
    }
    function de(t) {
      if (!Array.isArray(t))
        return process.env.NODE_ENV !== "production" && (arguments.length > 1 ? l(
          "Invalid arguments supplied to oneOf, expected an array, got " + arguments.length + " arguments. A common mistake is to write oneOf(x, y, z) instead of oneOf([x, y, z])."
        ) : l("Invalid argument supplied to oneOf, expected an array.")), j;
      function u(m, h, v, P, p) {
        for (var y = m[h], T = 0; T < t.length; T++)
          if (z(y, t[T]))
            return null;
        var x = JSON.stringify(t, function(B, o) {
          var X = Q(o);
          return X === "symbol" ? String(o) : o;
        });
        return new M("Invalid " + P + " `" + p + "` of value `" + String(y) + "` " + ("supplied to `" + v + "`, expected one of " + x + "."));
      }
      return V(u);
    }
    function ve(t) {
      function u(m, h, v, P, p) {
        if (typeof t != "function")
          return new M("Property `" + p + "` of component `" + v + "` has invalid PropType notation inside objectOf.");
        var y = m[h], T = K(y);
        if (T !== "object")
          return new M("Invalid " + P + " `" + p + "` of type " + ("`" + T + "` supplied to `" + v + "`, expected an object."));
        for (var x in y)
          if (s(y, x)) {
            var _ = t(y, x, v, P, p + "." + x, E);
            if (_ instanceof Error)
              return _;
          }
        return null;
      }
      return V(u);
    }
    function oe(t) {
      if (!Array.isArray(t))
        return process.env.NODE_ENV !== "production" && l("Invalid argument supplied to oneOfType, expected an instance of array."), j;
      for (var u = 0; u < t.length; u++) {
        var m = t[u];
        if (typeof m != "function")
          return l(
            "Invalid argument supplied to oneOfType. Expected an array of check functions, but received " + ne(m) + " at index " + u + "."
          ), j;
      }
      function h(v, P, p, y, T) {
        for (var x = [], _ = 0; _ < t.length; _++) {
          var B = t[_], o = B(v, P, p, y, T, E);
          if (o == null)
            return null;
          o.data && s(o.data, "expectedType") && x.push(o.data.expectedType);
        }
        var X = x.length > 0 ? ", expected one of type [" + x.join(", ") + "]" : "";
        return new M("Invalid " + y + " `" + T + "` supplied to " + ("`" + p + "`" + X + "."));
      }
      return V(h);
    }
    function pe() {
      function t(u, m, h, v, P) {
        return G(u[m]) ? null : new M("Invalid " + v + " `" + P + "` supplied to " + ("`" + h + "`, expected a ReactNode."));
      }
      return V(t);
    }
    function ie(t, u, m, h, v) {
      return new M(
        (t || "React class") + ": " + u + " type `" + m + "." + h + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + v + "`."
      );
    }
    function ue(t) {
      function u(m, h, v, P, p) {
        var y = m[h], T = K(y);
        if (T !== "object")
          return new M("Invalid " + P + " `" + p + "` of type `" + T + "` " + ("supplied to `" + v + "`, expected `object`."));
        for (var x in t) {
          var _ = t[x];
          if (typeof _ != "function")
            return ie(v, P, p, x, Q(_));
          var B = _(y, x, v, P, p + "." + x, E);
          if (B)
            return B;
        }
        return null;
      }
      return V(u);
    }
    function H(t) {
      function u(m, h, v, P, p) {
        var y = m[h], T = K(y);
        if (T !== "object")
          return new M("Invalid " + P + " `" + p + "` of type `" + T + "` " + ("supplied to `" + v + "`, expected `object`."));
        var x = S({}, m[h], t);
        for (var _ in x) {
          var B = t[_];
          if (s(t, _) && typeof B != "function")
            return ie(v, P, p, _, Q(B));
          if (!B)
            return new M(
              "Invalid " + P + " `" + p + "` key `" + _ + "` supplied to `" + v + "`.\nBad object: " + JSON.stringify(m[h], null, "  ") + `
Valid keys: ` + JSON.stringify(Object.keys(t), null, "  ")
            );
          var o = B(y, _, v, P, p + "." + _, E);
          if (o)
            return o;
        }
        return null;
      }
      return V(u);
    }
    function G(t) {
      switch (typeof t) {
        case "number":
        case "string":
        case "undefined":
          return !0;
        case "boolean":
          return !t;
        case "object":
          if (Array.isArray(t))
            return t.every(G);
          if (t === null || d(t))
            return !0;
          var u = g(t);
          if (u) {
            var m = u.call(t), h;
            if (u !== t.entries) {
              for (; !(h = m.next()).done; )
                if (!G(h.value))
                  return !1;
            } else
              for (; !(h = m.next()).done; ) {
                var v = h.value;
                if (v && !G(v[1]))
                  return !1;
              }
          } else
            return !1;
          return !0;
        default:
          return !1;
      }
    }
    function re(t, u) {
      return t === "symbol" ? !0 : u ? u["@@toStringTag"] === "Symbol" || typeof Symbol == "function" && u instanceof Symbol : !1;
    }
    function K(t) {
      var u = typeof t;
      return Array.isArray(t) ? "array" : t instanceof RegExp ? "object" : re(u, t) ? "symbol" : u;
    }
    function Q(t) {
      if (typeof t > "u" || t === null)
        return "" + t;
      var u = K(t);
      if (u === "object") {
        if (t instanceof Date)
          return "date";
        if (t instanceof RegExp)
          return "regexp";
      }
      return u;
    }
    function ne(t) {
      var u = Q(t);
      switch (u) {
        case "array":
        case "object":
          return "an " + u;
        case "boolean":
        case "date":
        case "regexp":
          return "a " + u;
        default:
          return u;
      }
    }
    function se(t) {
      return !t.constructor || !t.constructor.name ? k : t.constructor.name;
    }
    return F.checkPropTypes = O, F.resetWarningCache = O.resetWarningCache, F.PropTypes = F, F;
  }, Ie;
}
if (process.env.NODE_ENV !== "production") {
  var qr = sr(), Ur = !0;
  De.exports = Lr()(qr.isElement, Ur);
} else
  De.exports = Dr()();
var Nr = De.exports;
const $e = /* @__PURE__ */ Ar(Nr), Vr = ({ message: a = "Loading", spinStyle: S = "cp-round", opacity: E = 0.3, color: s = "white" }) => {
  const O = "cp-spinner " + S;
  return /* @__PURE__ */ ge.jsx("div", { className: "react-csspin-modal", style: { backgroundColor: s, opacity: E }, children: /* @__PURE__ */ ge.jsxs("div", { className: "react-csspin-box", children: [
    /* @__PURE__ */ ge.jsx("div", { className: O }),
    /* @__PURE__ */ ge.jsx("div", { style: { margin: "1rem" }, children: a })
  ] }) });
};
Vr.propTypes = {
  message: $e.string,
  spinStyle: $e.string,
  opacity: (a, S, E) => {
    if (S === "opacity" && a[S] > 1 && a[S] < 0)
      return new Error(`${S}속성의 값은 0~1 범위의 값만 허용합니다(at ${E}).`);
  },
  color: $e.string
};
export {
  Vr as ReactCsspin
};
